### PR TITLE
Add possibility to set up highlighten prefix and postfix for snippet

### PR DESCRIPTION
**Pull request by lavrd**
This PR adds a possibility to change highlight prefix and postfix for snippets.

### CODE REVIEW

File: src/snippet/mod.rs [hunk 1]
```diff
@@ -148,7 +167,12 @@ fn search_fragments<'a>(
 ///
 /// Takes a vector of `FragmentCandidate`s and the text.
 /// Figures out the best fragment from it and creates a snippet.
-fn select_best_fragment_combination(fragments: &[FragmentCandidate], text: &str) -> Snippet {
+fn select_best_fragment_combination(
+    fragments: &[FragmentCandidate],
+    text: &str,
+    highlighting_prefix: &str,
```
Comment (fulmicoton) [inline src/snippet/mod.rs]: the arguments are polution this function.

Can we remove them, add a setter on the `Snippet` object instead, and set the prefix/suffix on after the callsite of this function?

File: src/snippet/mod.rs [hunk 2]
```diff
@@ -50,35 +53,51 @@ impl FragmentCandidate {
...
-const HIGHLIGHTEN_PREFIX: &str = "<b>";
-const HIGHLIGHTEN_POSTFIX: &str = "</b>";
-
 impl Snippet {
-    /// Create a new, empty, `Snippet`
+    /// Create a new `Snippet`.
+    fn new(
+        fragment: &str,
+        highlighted: Vec<Range<usize>>,
+        highlighting_prefix: &str,
```
Comment (fulmicoton) [inline src/snippet/mod.rs]: See other comment:

Remove this from the new constructor and move that to an accessor to remove noise from select_best_fragment_combination

**Review by fulmicoton — DISMISSED** [review]
See comments

**Comment by lavrd** [comment]
Thank you for review! Did these fixes and reduced test.

File: src/snippet/mod.rs (hunk 2 above)
Comment (lavrd) [reply src/snippet/mod.rs]: Removed and moved to Snippet

File: src/snippet/mod.rs (hunk 1 above)
Comment (lavrd) [reply src/snippet/mod.rs]: Removed and moved to Snippet

**Comment by lavrd** [comment]
@fulmicoton hi, what the status with this feature?

**Comment by lavrd** [comment]
@PSeitz hey, can you help with review, please?

File: src/snippet/mod.rs [hunk 3]
```diff
@@ -101,6 +115,12 @@ impl Snippet {
     pub fn highlighted(&self) -> &[Range<usize>] {
         &self.highlighted
     }
+
+    /// Sets highlighted prefix and postfix.
+    pub fn set_highlighted_elements(&mut self, prefix: &str, postfix: &str) {
```
Comment (PSeitz) [inline src/snippet/mod.rs]:
```suggestion
    pub fn set_snippet_prefix_postfix(&mut self, prefix: &str, postfix: &str) {
```

File: src/snippet/mod.rs [hunk 4]
```diff
@@ -55,17 +58,28 @@ impl FragmentCandidate {
 pub struct Snippet {
     fragment: String,
     highlighted: Vec<Range<usize>>,
+    highlighting_prefix: String,
```
Comment (PSeitz) [inline src/snippet/mod.rs]:
```suggestion
    snippet_prefix: String,
```

File: src/snippet/mod.rs [hunk 5]
```diff
@@ -55,17 +58,28 @@ impl FragmentCandidate {
...
+    highlighting_postfix: String,
```
Comment (PSeitz) [inline src/snippet/mod.rs]:
```suggestion
    snippet_postfix: String,
```

**Review by PSeitz — APPROVED** [review]

**Comment by PSeitz** [comment]
@lavrd Thanks!

**Pull request merged**